pkg/cmd/roachtest/test_registry.go

```diff
@@ -127,6 +128,25 @@ func (r *testRegistryImpl) prepareSpec(spec *registry.TestSpec) error {
 	}
 	spec.Tags = append(spec.Tags, "owner-"+string(spec.Owner))
 
+	// At the time of writing, we expect the roachtest job to finish within 24h.
+	// Since each individual test may not be scheduled until a few hours in,
+	// individual tests should expect to take "less time". Longer-running
+	// tests require the weekly tag.
+	const maxTimeout = 18 * time.Hour
+	if spec.Timeout > maxTimeout {
+		var weekly bool
+		for _, tag := range spec.Tags {
+			if tag == "weekly" {
+				weekly = true
+			}
+		}
+		if !weekly {
+			return fmt.Errorf(
+				"timeout %s for %s exceeds the maximum allowed timeout of %s", spec.Timeout, spec.Name, maxTimeout,
```

Comment:
The rest of the errors from this function are in the form of `("%s: timeout ..", spec.Name...)` if we wanted to stay consistent.